dask/array/core.py

```diff
@@ -2842,12 +2842,10 @@ def where(condition, x=None, y=None):
     x = broadcast_to(x, shape).astype(dtype)
     y = broadcast_to(y, shape).astype(dtype)
 
-    if isinstance(condition, (bool, np.bool8)):
-        if condition:
-            return x
-        else:
-            return y
+    if np.isscalar(condition):
+        return x if condition else y
     else:
+        condition = asarray(condition).astype('bool')
         return choose(condition, [y, x])
```

Comment:
Is there a particular reason to do things this way rather than `elemwise(np.where, condition, x, y)`? I'm pretty new to dask so I don't know if there is a subtlety that makes that not work in all cases, but `np.where` is a lot faster than `np.choose`, even before one considers the extra time and memory cost of the boolean cast.
```
a=np.zeros(100000000)
b=np.ones(100000000)
c=np.random.randint(0, 2, 100000000)
%timeit np.where(c, a, b)
1 loop, best of 3: 571 ms per loop
%timeit np.choose(c, [b, a])
1 loop, best of 3: 1.24 s per loop
```